### PR TITLE
Support Open-Generic-Types in 'nameof' expressions.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: docs/features/open-generic-names-in-nameof.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+```C#
+class KeyValuePair<A,B> 
+{
+}
+class G<T,U>
+{
+    public KeyValuePair<T,U> AProperty { get; }
+}
+class D<T> : G<T, int>
```
Comment (sharwell) [inline docs/features/open-generic-names-in-nameof.md]: :exclamation: The `T` defined here should be given a new name to eliminate ambiguity in referencing `G<T, int>` in the text below.

**Comment by CyrusNajmabadi** [comment]
@sharwell Thanks!  I've made that change.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/mac/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/mac/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/dbg/eta please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by jmarolf** [comment]
@CyrusNajmabadi could you merge all the changes from future into features/openGenericNameInNameof before submitting this PR?

**Comment by gafter** [comment]
@CyrusNajmabadi This PR is getting ancient. What is the LDM status?

**Comment by gafter** [comment]
@CyrusNajmabadi Closing the PR; we defer to the LDM for a decision.

**Pull request closed without merging**